tests/rptest/scale_tests/high_throughput_test.py

```diff
@@ -0,0 +1,928 @@
+        self.logger.info(f"Restarting node {node_str} for {restart_timeout} s")
+        self.redpanda.start_node(node, timeout=600)
+        wait_until(self.redpanda.healthy,
+                   timeout_sec=restart_timeout,
+                   backoff_sec=1)
+
+    @cluster(num_nodes=5, log_allow_list=NOS3_LOG_ALLOW_LIST)
+    def test_disrupt_cloud_storage(self):
+        """
+        Make segments replicate to the cloud, then disrupt S3 connectivity
+        and restore it
+        """
+        self.setup_cluster(
+            # Segments should go into the cloud at a reasonable rate,
+            # that's why it is smaller than it should be
+            segment_bytes=int(self.scaled_segment_size / 2),
+            retention_local_bytes=2 * self.scaled_segment_size,
+        )
+
+        try:
+            producer = KgoVerifierProducer(
+                self.test_context,
+                self.redpanda,
+                self.topic_name,
+                msg_size=128 * 1024,
+                msg_count=5 * 1024 * 1024 * 1024 * 1024,
+                rate_limit_bps=self.scaled_data_bps,
+                custom_node=[self.preallocated_nodes[0]])
+            producer.start()
+            wait_until(lambda: producer.produce_status.acked > 10000,
+                       timeout_sec=60,
+                       backoff_sec=1.0)
+
+            # S3 up -> down -> up
+            self.stage_block_s3()
+
+        finally:
+            producer.stop()
+            producer.wait(timeout_sec=600)
+            self.free_preallocated_nodes()
+
+    def _cloud_storage_no_new_errors(self, redpanda, logger=None):
+        num_errors = redpanda.metric_sum(
+            "redpanda_cloud_storage_errors_total",
+            metrics_endpoint=MetricsEndpoint.PUBLIC_METRICS)
+        increase = (num_errors -
+                    self.last_num_errors) if self.last_num_errors > 0 else 0
+        self.last_num_errors = num_errors
+        if logger:
+            logger.info(
+                f"Cluster metrics report {increase} new cloud storage errors ({num_errors} overall)"
+            )
+        return increase == 0
+
+    def stage_block_s3(self):
+        self.logger.info(f"Getting the first 100 segments into the cloud")
+        wait_until(lambda: nodes_report_cloud_segments(self.redpanda, 100),
+                   timeout_sec=120,
+                   backoff_sec=5)
+        self.logger.info(f"Blocking S3 traffic for all nodes")
+        self.last_num_errors = 0
+        with firewall_blocked(self.redpanda.nodes, self.s3_port):
+            # wait for the first cloud related failure + one minute
+            wait_until(lambda: not self._cloud_storage_no_new_errors(
+                self.redpanda, self.logger),
+                       timeout_sec=600,
+                       backoff_sec=10)
+            time.sleep(60)
+        # make sure nothing is crashed
+        wait_until(self.redpanda.healthy, timeout_sec=60, backoff_sec=1)
```

Comment:
nit: should this be an immediate check rather than a wait? Does become healthy in between this?

Reply:
According to the logic of the test, it should be healthy already. However it is always safer to retry (what if the endpoint timeouts once or twice?).